### PR TITLE
add a `list` command

### PR DESCRIPTION
**Pull request by logikal**
I can never remember the names of the profiles I have configured, so this command will list them for you.

Caveat: It assumes you only care about profiles that are assuming from another role; it only lists profiles that have a source_profile.

### CODE REVIEW

**Comment by logikal** [comment]
@dfuentes Is this a feature ya'll would like?

**Comment by nickatsegment** [comment]
@logikal 	Yeah, I think it could be useful. Let me take a look.

File: cmd/list.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+	Use:   "list",
+	Short: "list will show you the profiles currently configured",
+	RunE:  listRun,
+}
+
+func init() {
+	RootCmd.AddCommand(listCmd)
+}
+
+func listPre(cmd *cobra.Command, args []string) {
```
Comment (nickatsegment) [inline cmd/list.go]: Dead code?

File: cmd/list.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+	w.Flush()
+
+	if analyticsEnabled && analyticsClient != nil {
+		analyticsClient.Enqueue(analytics.Track{
+			UserId: username,
+			Event:  "Listed Profiles",
+			Properties: analytics.NewProperties().
+				Set("backend", backend).
+				Set("aws-okta-version", version).
+				Set("profile_count", len(profiles)).
```
Comment (nickatsegment) [inline cmd/list.go]: I think this should be `profile-count`, to standardize on dashes being the word separator. @systemizer 	may have opinions.

**Comment by logikal** [comment]
Thanks, @nickatsegment! I've fixed up the feedback and updated from master, squashed it all up, and think it's ready to go now. 🎅

**Comment by logikal** [comment]
Anything else needed here, @nickatsegment ?

**Comment by nickatsegment** [comment]
Sorry for the delay!

File: cmd/list.go [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+	w := new(tabwriter.Writer)
+	w.Init(os.Stdout, 0, 8, 2, '\t', 0)
+	fmt.Fprintln(w, "profile\tarn\tsource_role\t")
+	fmt.Fprintln(w, "---\t---\t---\t")
+	for profile, v := range profiles {
+		if role, exist := v["role_arn"]; exist {
+			if src, exist := v["source_profile"]; exist {
+				s := fmt.Sprintf("%s\t%s\t%s\t", profile, role, src)
+				fmt.Fprintln(w, s)
```
Comment (nickatsegment) [inline cmd/list.go]: Just use `fmt.Fprintf` directly?

File: cmd/list.go [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+		return err
+	}
+
+	w := new(tabwriter.Writer)
+	w.Init(os.Stdout, 0, 8, 2, '\t', 0)
+	fmt.Fprintln(w, "profile\tarn\tsource_role\t")
+	fmt.Fprintln(w, "---\t---\t---\t")
+	for profile, v := range profiles {
+		if role, exist := v["role_arn"]; exist {
+			if src, exist := v["source_profile"]; exist {
```
Comment (nickatsegment) [inline cmd/list.go]: In the case where there is no `source_profile`, no line gets printed. This is the case in my config, so I get no results from the whole command :( I wonder if you could use `lib/config.go`'s `sourceProfile` function?

**Comment by logikal** [comment]
Thanks, @nickatsegment! 

I fixed this to display profiles that don't have a source_profile, and also have it sort by `profile_name` so it gives the same order on each run.

**Comment by nickatsegment** [comment]
🚢

**Pull request merged**